client/lib/domains/get-fixed-domain-search.js

```diff
@@ -7,5 +7,6 @@ export function getFixedDomainSearch( domainName ) {
 		.replace( /^www[0-9]?\./, '' )
 		.replace( /\/$/, '' )
 		.replace( /_/g, '-' )
+		.replace( /[^a-z0-9-]/, '' )
 		.replace( /^\.+|\.+$/, '' );
```

Comment:
Your `replace` will take care of these.

```suggestion
		.replace( /[^a-z0-9-]/, '' )
```